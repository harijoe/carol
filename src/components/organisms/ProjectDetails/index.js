import React, { PropTypes } from 'react'
import styled from 'styled-components'

import { Project, Loading } from 'components'

const Wrapper = styled.div`
  & > * {
    margin: 1em;
  }
`

const ProjectDetails = ({ details, loading, ...props }) => (
  <Wrapper {...props}>
    <Loading loading={loading && details === null}>
      {
        details != null &&
        <Project
          items={details}
          full
        />
      }
    </Loading>
  </Wrapper>
)

ProjectDetails.propTypes = {
  details: PropTypes.object,
  loading: PropTypes.bool,
}

export default ProjectDetails
