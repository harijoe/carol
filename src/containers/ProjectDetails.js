import React, { PropTypes, Component } from 'react'
import { connect } from 'react-redux'

import { fromProject, fromStatus } from 'store/selectors'
import { projectDetails, PROJECT_DETAILS, setToken } from 'store/actions'
import { ProjectDetails } from 'components'

class ProjectDetailsContainer extends Component {
  static propTypes = {
    details: PropTypes.object,
    loading: PropTypes.bool,
    request: PropTypes.func.isRequired,
  }

  componentWillMount() {
    this.props.request()
  }

  render() {
    const { details, loading } = this.props

    return <ProjectDetails {...{ details, loading }} />
  }
}

const mapStateToProps = (state, { id }) => ({
  details: fromProject.getDetails(state, id),
  loading: fromStatus.isLoading(state, PROJECT_DETAILS.prefix),
})

const mapDispatchToProps = (dispatch, { id }) => ({
    // @TODO: when #191 is finished
  request: () => {
    setToken(dispatch).then(() => {
      dispatch(projectDetails.request(id))
    })
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(ProjectDetailsContainer)
