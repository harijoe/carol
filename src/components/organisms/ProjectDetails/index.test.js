import React from 'react'
import { shallow } from 'enzyme'
import ProjectDetails from './'
import mockIntl from '../../../../test/intlMock'

const details = { id: '1', name: 'project1', trade: 'test' }
const wrap = (props = {}) => shallow(mockIntl(<ProjectDetails details={details} {...props} />))

it('renders ProjectDetails', () => {
  expect(wrap()).toMatchSnapshot()
})
