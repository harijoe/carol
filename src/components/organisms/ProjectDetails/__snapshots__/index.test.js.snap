// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProjectDetails 1`] = `
<InjectTranslate(ProjectDetails)
  details={
    Object {
      "id": "1",
      "name": "project1",
      "trade": "test",
    }
  }
/>
`;
